lxd/instance/drivers/driver_qemu.go

```diff
@@ -3181,10 +3182,70 @@ func (d *qemu) addDriveConfig(bootIndexes map[string]int, driveConf deviceConfig
 		"read-only": false,
 	}
 
+	var secret string
+
 	// If driver is "file", QEMU requires the file to be a regular file.
 	// However, if the file is a character or block device, driver needs to be set to "host_device".
 	if isBlockDev {
 		blockDev["driver"] = "host_device"
+	} else if isRBDImage {
+		blockDev["driver"] = "rbd"
+		// Expected format is rbd:poolname/volumename:<options>
+		// where options is key=value:...
+		fields := strings.SplitN(driveConf.DevPath, ":", 3)
+
+		if len(fields) != 3 {
+			return nil, fmt.Errorf("Unexpected devPath rbd format %q for drive %q", driveConf.DevPath, driveConf.DevName)
+		}
+
+		opts := fields[2]
+		fields = strings.Split(fields[1], "/")
+
+		// Driver and pool name arguments can be ignored as CephGetRBDImageName doesn't need them.
+		vol := storageDrivers.NewVolume(nil, "", storageDrivers.VolumeTypeCustom, storageDrivers.ContentTypeBlock, project.StorageVolume(d.project, fields[1]), nil, nil)
+		rbdImageName := storageDrivers.CephGetRBDImageName(vol, "", false)
+
+		userName := "admin"
+		clusterName := ""
+
+		for _, option := range strings.Split(opts, ":") {
+			fields := strings.Split(option, "=")
+
+			if fields[0] == "id" && fields[1] != "" {
+				userName = fields[1]
+			} else if fields[0] == "conf" && fields[1] != "" {
+				baseName := filepath.Base(fields[1])
+				clusterName = strings.TrimSuffix(baseName, ".conf")
```

Comment:
For my own knowledge, is this always correct that the name of the config file used indicates the cluster name?

CC @stgraber